src/app/Store/canvas.state.ts

```diff
@@ -0,0 +1,13 @@
+import { fabric } from 'fabric';
+
+export interface Canvas {
+    canvas: string;
+}
+
+export interface State {
+    canvasArray: Canvas[];
+}
+
+export const initialState: State = {
+    canvasArray: [{ canvas: JSON.stringify(new fabric.Canvas('canvas')) }],
+};
```

Comment:
Store should ideally contain only the present objects on the canvas. Not the history of all the states of the canvas over time. Refer [Three Principles of Redux](https://redux.js.org/understanding/thinking-in-redux/three-principles) to get a few guidelines on how we should use state management.